### PR TITLE
GNUInstall Dir enhancements

### PR DESCRIPTION
**Pull request by arvindcheru**
- Proposed changes for GNUInstall dir enhancements

### CODE REVIEW

**Comment by cgmb** [comment]
Thanks! That looks great. Please update to remove
```cmake
# Force library install path to lib (CentOS 7 defaults to lib64)
set(CMAKE_INSTALL_LIBDIR "lib" CACHE INTERNAL "Installation directory for libraries" FORCE)
```
now that rocm-cmake is handling that behaviour.

**Review by cgmb — APPROVED** [review]
LGTM.

**Review by jzuniga-amd — APPROVED** [review]

**Review by tfalders — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -207,8 +207,8 @@ endif()
 
 set(CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION
   "\${CPACK_PACKAGING_INSTALL_PREFIX}"
-  "\${CPACK_PACKAGING_INSTALL_PREFIX}/include"
-  "\${CPACK_PACKAGING_INSTALL_PREFIX}/lib"
+  "\${CPACK_PACKAGING_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}"
```
Comment (Mystro256) [inline CMakeLists.txt:207]: ${CPACK_PACKAGING_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}

can be replaced with:

${CPACK_INSTALL_FULL_INCLUDEDIR}'

Same for "CPACK_INSTALL_FULL_LIBDIR"

**Comment by Mystro256** [comment]
Please note that 

    ${CMAKE_INSTALL_FULL_*DIR} is defined as ${CPACK_INSTALL_PREFIX}/${CMAKE_INSTALL_*DIR}

I'm assuming that

    ${CPACK_PACKAGING_INSTALL_PREFIX} = ${CPACK_INSTALL_PREFIX}

File: CMakeLists.txt (hunk 1 above)
Comment (cgmb) [reply CMakeLists.txt:207]: The `CPACK_PACKAGING_INSTALL_PREFIX` and `CMAKE_INSTALL_PREFIX` are different, though. The `CMAKE_INSTALL_PREFIX` is `<builddir>/rocsolver-install` but `CPACK_PACKAGING_INSTALL_PREFIX` is `/opt/rocm`.

**Comment by cgmb** [comment]
> I'm assuming that
> 
> ```
> ${CPACK_PACKAGING_INSTALL_PREFIX} = ${CPACK_INSTALL_PREFIX}
> ```

Sorry, but this assumption is wrong. I don't think your suggestion is correct.

**Comment by Mystro256** [comment]
> > I'm assuming that
> > ```
> > ${CPACK_PACKAGING_INSTALL_PREFIX} = ${CPACK_INSTALL_PREFIX}
> > ```
> 
> Sorry, but this assumption is wrong. I don't think your suggestion is correct.

Ok thanks, please ignore my comment then

**Pull request merged**